tmt/result.py

```diff
@@ -113,26 +124,20 @@ def __init__(
                 self.ids[key] = test.node.get(key)
             interpret = ResultInterpret(test.result) if test.result else ResultInterpret.RESPECT
         else:
-            try:
-                self.ids = data['ids']
-            except KeyError:
-                self.ids = {}
+            self.ids = data.ids
             interpret = ResultInterpret.RESPECT
 
         # Check for valid results
-        try:
-            self.result = ResultOutcome(data['result'])
-        except KeyError:
-            raise tmt.utils.SpecificationError("Missing test result.")
-        except ValueError:
-            raise tmt.utils.SpecificationError(
-                f"Invalid result '{data['result']}'.")
+        # try:
+        self.result = data.result
+        # except KeyError:
+        #     raise tmt.utils.SpecificationError("Missing test result.")
+        # except ValueError:
+        #     raise tmt.utils.SpecificationError(
+        #         f"Invalid result '{data['result']}'.")
 
         # Convert log into list if necessary
-        try:
-            self.log = tmt.utils.listify(data['log'])
-        except KeyError:
-            self.log = []
+        self.log = tmt.utils.listify(data.log)
```

Comment:
This shouldn't be needed as `data.log` should already be a list. It's up to caller to provide a list of logs.